pre-commit-hook.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Define the pre-commit hook file path
+HOOKS_DIR=".git/hooks"
+PRE_COMMIT_FILE="$HOOKS_DIR/pre-commit"
+
+# Create the hooks directory if it doesn't exist
+mkdir -p $HOOKS_DIR
+
+# Write the pre-commit hook script
+cat <<'EOF' > $PRE_COMMIT_FILE
+#!/bin/bash
+
+# Navigate to the subfolder
+PROJECT_DIR="website"
+cd $PROJECT_DIR || exit 1
+
+# Run formatting and stage changes
+echo "Running Prettier formatting..."
+pnpm run format
+if [ $? -ne 0 ]; then
+  echo "Prettier formatting failed. Aborting commit."
+  exit 1
+fi
+
+# Stage any changes made by the formatter
+echo "Staging changes made by Prettier..."
+git add .
+
+# Run linting with auto-fix and stage changes
+echo "Running ESLint with auto-fix..."
+pnpm run lint --fix
+if [ $? -ne 0 ]; then
+  echo "ESLint check failed. Aborting commit."
+  exit 1
+fi
+
+# Stage any changes made by the linter
+echo "Staging changes made by ESLint..."
+git add .
+
```

Comment:
This hook doesn't take the following into account: there are 3 kinds of files in Git.

- Tracked and staged
- Tracked and unstaged
- Untracked

The same file can also be `tracked and staged` and `tracked and unstaged` at the same time (using `git add --patch`).
Therefore, this hook should only run on the `tracked and staged` patches.
To do that, you need to use commits and the stash to achieve the desired effect.

1. Before all linting, do `git stash push -S --quiet`. This ensures all staged changes (the one linting will be applied on) are stashed.
2. Then stash all other changed using `git stash push --include-untracked --quiet`.
3. Then pop the entry made in the first step using `git stash pop stash@{1}`.
4. Run all linting using `pnpm run lint --fix`.
5. Add all files using `git add .`
6. Then, unstash the changes that were stashed in 2 using `git stash pop`.
7. Then, proceed with the commit.

Please take into account that some `git stash pop` can cause conflicts and, thus, can cause this whole process to abort. What you should do in this case, for safety, is to revert to the initial state.

This can be done by creating an initial stash entry using `git stash push -u` and unstashing at the end using `git stash pop --index`.